### PR TITLE
Update long haul resource generation script

### PR DESCRIPTION
**Pull request by abhipsaMisra**

### CODE REVIEW

File: e2e/LongHaul/prerequisites/LongHaulSetup.ps1 [hunk 1]
```diff
@@ -22,7 +22,6 @@ $startTime = (Get-Date)
 ########################################################################################################
 
 $ErrorActionPreference = "Stop"
-$WarningActionPreference = "Continue"
```
Comment (abhipsaMisra) [inline e2e/LongHaul/prerequisites/LongHaulSetup.ps1:25]: unused

File: e2e/LongHaul/prerequisites/LongHaulSetup.ps1 [hunk 2]
```diff
@@ -216,13 +213,6 @@ $keyVaultName = az deployment group show -g $ResourceGroup -n $deploymentName --
 $iotHubName = az deployment group show -g $ResourceGroup -n $deploymentName --query 'properties.outputs.hubName.value' --output tsv
 $instrumentationKey = az deployment group show -g $ResourceGroup -n $deploymentName --query 'properties.outputs.instrumentationKey.value' --output tsv
 
-##################################################################################################################################
-# Fetch the iothubowner policy details.
-##################################################################################################################################
-
-$iothubownerSasPolicy = "iothubowner"
-$iothubownerSasPrimaryKey = az iot hub policy show --hub-name $iotHubName --name $iothubownerSasPolicy --query 'primaryKey'
```
Comment (abhipsaMisra) [inline e2e/LongHaul/prerequisites/LongHaulSetup.ps1:224]: unused

File: e2e/LongHaul/prerequisites/LongHaulSetup.ps1 [hunk 3]
```diff
@@ -232,7 +222,7 @@ $longhaulDeviceModuleId = "LongHaulDeviceModule1"
 $longhaulEdgeDeviceId = "LongHaulEdgeDevice1"
 $longhaulEdgeModuleId = "LongHaulEdgeModule1"
 $longhaulDevice = az iot hub device-identity list -g $ResourceGroup --hub-name $iotHubName --query "[?deviceId=='$longhaulDeviceId'].deviceId" --output tsv
-$longhaulDeviceModule = az iot hub module-identity list -g $ResourceGroup --hub-name $iotHubName --query "[?moduleId=='$longhaulDeviceModuleId'].moduleId" --output tsv
+$longhaulDeviceModule = az iot hub module-identity list -g $ResourceGroup --hub-name $iotHubName --device-id $longhaulDeviceId --query "[?moduleId=='$longhaulDeviceModuleId'].moduleId" --output tsv
```
Comment (abhipsaMisra) [inline e2e/LongHaul/prerequisites/LongHaulSetup.ps1:225]: "module-identity list" requires the associated deviceID. Similar to the "longhaulEdgeModule" script below.

File: e2e/LongHaul/prerequisites/LongHaulSetup.ps1 [hunk 4]
```diff
@@ -295,11 +285,11 @@ foreach ($kvp in $keyvaultKvps.GetEnumerator())
 # Creating a file to run to load environment variables
 ############################################################################################################################
 
-$loadScriptDir = Join-Path $PSScriptRoot "..\..\..\.." -Resolve
-$loadScriptName = "Load-$keyVaultName.ps1";
+$loadScriptDir = Join-Path $PSScriptRoot "..\..\.." -Resolve
+$loadScriptName = "Load-LongHaul-$keyVaultName.ps1";
 Write-Host "`nWriting environment loading file to $loadScriptDir\$loadScriptName.`n"
 $file = New-Item -Path $loadScriptDir -Name $loadScriptName -ItemType "file" -Force
-Add-Content -Path $file.PSPath -Value "$loadScriptDir\azure-iot-sdk-csharp\e2e\Tests\prerequisites\LoadEnvironmentVariablesFromKeyVault.ps1 -SubscriptionId $SubscriptionId -KeyVaultName $keyVaultName"
```
Comment (abhipsaMisra) [inline e2e/LongHaul/prerequisites/LongHaulSetup.ps1:302]: People might have renamed their root folder name, so we shouldn't take a dependency on that

**Review by tmahmood-microsoft — APPROVED** [review]

**Pull request merged**